src/bootstrap/replication/service.rs

```diff
@@ -1119,64 +1185,64 @@ impl ReplicatorBootstrap {
     ) -> Result<(), BootstrapError> {
         info!("Writing final block and shard chunks to database...");
 
+        // Write shard-0 block
```

Comment:
Why did we need to make this change? What's the difference in behavior and why was it necessary? It's not obvious to me why the existing code is wrong.